### PR TITLE
Add webfaction modules

### PR DESCRIPTION
**Pull request by quentinsf**
These modules, which provide a wrapper around the webfaction API and allow you to create sites, apps, databases, domains and mailboxes, have proven fairly popular in my own github space, so I thought I should try and move them somewhere a bit more official!

Hope they're useful?

All the best, 
Quentin

P.S. I wasn't sure what to put in the version_added documentation field, so at present they say 1.99.

### CODE REVIEW

File: cloud/webfaction/webfaction_app.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        required: true
+
+    login_password:
+        description:
+            - The webfaction password to use
+        required: true
+'''
+
+import xmlrpclib
+from ansible.module_utils.basic import *
```
Comment (sivel) [inline cloud/webfaction/webfaction_app.py]: The module_utils import should be located just before the call to `main()`

**Comment by quentinsf** [comment]
Thanks guys - have made changes as per your suggestions..

File: cloud/webfaction/webfaction_app.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import xmlrpclib
+
+webfaction = xmlrpclib.ServerProxy('https://api.webfaction.com/')
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(required=True),
+            state = dict(required=False, default='present'),
```
Comment (resmo) [inline cloud/webfaction/webfaction_app.py]: add choises=['present', 'absent'] here like you did in doc.

File: cloud/webfaction/webfaction_app.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+# Quentin Stafford-Fraser 2015
+
+DOCUMENTATION = '''
+---
+module: webfaction_app
+short_description: Add or remove applications on a Webfaction host
+description:
+    - Add or remove applications on a Webfaction host.  Further documentation at http://github.com/quentinsf/ansible-webfaction.
+author: Quentin Stafford-Fraser
+version_added: 2.0
```
Comment (resmo) [inline cloud/webfaction/webfaction_app.py]: use "2.0" for versions

File: cloud/webfaction/webfaction_domain.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        description:
+            - The webfaction account to use
+        required: true
+
+    login_password:
+        description:
+            - The webfaction password to use
+        required: true
+'''
+
```
Comment (resmo) [inline cloud/webfaction/webfaction_domain.py:68]: missing examples?

File: cloud/webfaction/webfaction_app.py (hunk 2 above)
Comment (quentinsf) [inline cloud/webfaction/webfaction_app.py]: Thanks resmo - now added

File: cloud/webfaction/webfaction_domain.py (hunk 4 above)
Comment (quentinsf) [inline cloud/webfaction/webfaction_domain.py:68]: Added in latest commit.

File: cloud/webfaction/webfaction_app.py (hunk 3 above)
Comment (quentinsf) [inline cloud/webfaction/webfaction_app.py]: Now changed to strings.

**Comment by gregdek** [comment]
Did last checkins break something? It fails Travis build, but I don't have enough context to know why...

**Comment by quentinsf** [comment]
I don't think it's an issue with the webfaction stuff - the limited build errors I can see are all elsewhere.

I obviously don't have a way to run the same tests here, but the modules do at least compile cleanly for me, and if I copy them into an ansible/library/cloud directory with appropriate renaming, the docs also build correctly.

**Comment by bcoca** [comment]
so we are just missing the GPL header at this point, everything else looks OK

**Comment by quentinsf** [comment]
Have added GPL headers, fixed a missing param in the documentation, and made one other change which makes database creation/deletion more symmetric/repeatable.

**Comment by bcoca** [comment]
@quentinsf 
the login/password functionality and documentation can centralized, for examples look at 
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py
https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/module_docs_fragments/aws.py

**Pull request merged**